orix/quaternion/orientation.py

```diff
@@ -375,6 +375,73 @@ def get_distance_matrix(
 
         return angles
 
+    @classmethod
+    def from_align_vectors(
+        cls,
+        to_: Miller,
+        from_: Miller,
+        weights: Optional[np.ndarray] = None,
+        return_rmsd: bool = False,
+        return_sensitivity: bool = False,
+    ) -> Misorientation:
+        """ "Return an estimated misorientation to optimally align two sets of vectors, one in each crystal."
+
+        This method wraps :meth:`scipy.spatial.transform.Rotation.align_vectors`,
+        see that method for further explanations of parameters and returns.
+
+        Parameters
+        ----------
+        to_
+            Directions in the other crystal.
+        from_
+            Directions in the initial crystal.
+        weights
+            The relative importance of the different vectors.
+        return_rmsd
+            Whether to return the root mean square distance (weighted)
+            between ``to_`` and ``from_`` after alignment.
+        return_sensitivity
+            Whether to return the sensitivity matrix.
+
+        Returns
+        -------
+        estimated misorientation
+            Best estimate of the ``misorientation`` that transforms ``from_`` to ``to_``.
+            The symmetry of the ``misorientation`` is inferred from the phase of ``to_`` and ``from_``, if given.
+        rmsd
+            Returned when ``return_rmsd=True``.
+        sensitivity
+            Returned when ``return_sensitivity=True``.
+
+        Examples
+        -------
+        >>> from orix.quaternion import Misorientation
+        >>> from orix.vector import Vector3d, Miller
+        >>> crystal1 = Miller([[1, 0, 0], [0, 1, 0]])
+        >>> crystal2 = Miller([[1, 0, 0], [0, 0, 1]])
+        >>> Misorientation.from_align_vectors(crystal1, crystal2)
+        Misorientation (1,) 1
+        [[-0.7071  0.7071  0.      0.    ]]
+        """
+        if not isinstance(to_, Miller) or not isinstance(from_, Miller):
+            raise ValueError(
+                "Arguments to_ and from_ must both be of type Miller, "
+                f"but are of type {type(to_)} and {type(from_)}."
+            )
+
+        out = super().from_align_vectors(
+            to_=to_,
+            from_=from_,
+            weights=weights,
+            return_rmsd=return_rmsd,
+            return_sensitivity=return_sensitivity,
+        )
+        out = list(out)
+        if hasattr(to_.phase, "point_group") and hasattr(from_.phase, "point_group"):
```

Comment:
This will always be true, so it must be

```suggestion
        if to_.phase.point_group and from_.phase.point_group:
```

Reply:
A miller object can me initialized without a phase argument.

```
from orix.vector import Miller
m = Miller([[1,0,0]]) #this runs fine
print(m.phase.point_group) #error appears here
```
output
```
AttributeError: 'NoneType' object has no attribute 'point_group'
```

And so we could get errors raised if we use `if to_.phase.point_group:`. But the code above might be a bug with the Miller object?

Reply:
You are right. To avoid too many checks, I suggest to just use

```python
try:
    mori = (from_.phase.point_group, to_.phase.point_group)
except (AttributeError, ValueError):
    pass
```

`AttributeError` is raised if the `Miller.phase == None`, and `ValueError` is raised if both `Miller.phase.point_group == None`.

It's not a bug, but this shows that a warning should be raised when passing `Miller([1, 0, 0])`, because then it is assumed that this vector is expressed in terms of the sample reference frame (cartesian coordinates, `xyz`) and not the crystal. This behavior is described in [the documentation](https://orix.readthedocs.io/en/stable/reference/generated/orix.vector.Miller.html). In most cases, if not all, the user should pass the vector as `uvw` or `hkl` etc. Having the possibility to just pass a vector without specifying the vector type or phase is necessary so that methods inherited from `Vector3d` and `Object3d`, where `data` and nothing else is passed on, works. Inherited methods in `Miller` is not handled perfectly yet, and this issue https://github.com/pyxem/orix/issues/367 was raised to improve this.